Automattic/sensei
# Add sensei home rest API mock.

Fixes #5676 

### Changes proposed in this Pull Request
* Create `/sensei-internal/v1/home` endpoint to return all the data needed by new Sensei Home screen.
* This is only an initial pull request to discuss the interface to be consumed from Sensei Home frontend. Implementation will come later probably in separate tickets.
* Based on 1-gh-Automattic/senseilms-com-plugins

### Testing instructions
- Check the structures fit all requirements in the Home i2 iteration P2.
- Check pull request for comments with further questions/doubts.

### Discussion
- ~Should we return from the backend the entries for the Help section? They could be frontend only since we won't have a place to persist those entries and will be basically hardcoded in the backend.~ Yes, returning from the backend so we can customise them based on plugin installed and extensions.
- We might end up separating the Tasks List into a separate endpoint or maybe even other parts too. Any ideas on granularity here?
- ~Task list also contains an image that changes based on the current step. Should we return the image url from the backend or maybe let the frontend decide the image by checking which is the first "not done" task from the list? The second part would mean returning some ID so frontend can identify the step and map to a correct image. Probably it's easier to return the image url from backend. WDYT?~ For the moment returning an optional image on every task.
